### ISSUE TITLE
Setting `**dict[str, int]` to `**kwargs` gets a wrong error message, saying `argument` instead of `parameter`

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20679]
*Memo:
- mypy test.py
- mypy 1.19.1
- Python 3.14.0

Setting `**dict[str, int]` to `**kwargs` gets the wrong error message, saying `argument` instead of `parameter` as shown below:

```python
from typing import Unpack

def func(**kwargs: Unpack[dict[str, int]]) -> None: ... # Error
```

> error: Unpack item in ** argument must be a TypedDict

So, the error message should be as shown below:

> error: Unpack item in ** parameter must be a TypedDict

**Comment by A5rocks** [comment]
Hm, I think this is already correct? Given what happened already re: type args I'm not completely confident, but last time @hauntsaninja distinguished between calling (argument) and definition (parameter). This is definitely the latter!

**Comment by A5rocks** [comment]
Wait sorry, I misread the bug report. This seems fine...
